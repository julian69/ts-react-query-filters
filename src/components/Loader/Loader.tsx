import { Oval } from 'react-loader-spinner';

const Loader = () => (
  <Oval
    visible={true}
    height="50"
    width="50"
    color="#000"
    secondaryColor="#828282"
    ariaLabel="oval-loading"
    wrapperStyle={{}}
    wrapperClass=""
  />);

export default Loader;
