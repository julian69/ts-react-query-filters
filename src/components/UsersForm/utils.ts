import { IFilters } from "@/types/users";

export const deserializeUrl = (params: URLSearchParams): IFilters => ({
  name: params.get('name') || '',
  email: params.get('email') || '',
  company: params.get('company') || '',
  zipcode: params.get('zipcode') || '',
});


