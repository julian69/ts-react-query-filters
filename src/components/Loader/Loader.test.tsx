import '@testing-library/jest-dom';
import { render } from '@/test/utils';
import { describe, expect, it } from 'vitest';

import Loader from './Loader';

describe('<Loader />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Loader />);
    expect(asFragment()).toMatchSnapshot();
  });
});