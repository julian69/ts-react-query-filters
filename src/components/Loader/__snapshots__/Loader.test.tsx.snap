// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Loader /> > should render correctly 1`] = `
<DocumentFragment>
  <div
    aria-busy="true"
    aria-label="oval-loading"
    class="sc-beySPh gZmPOF"
    data-testid="oval-loading"
    role="progressbar"
  >
    <svg
      data-testid="oval-svg"
      height="50"
      stroke="#000"
      viewBox="-20 -20 42 42"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <g
          data-testid="oval-secondary-group"
          stroke-width="2"
          transform="translate(1 1)"
        >
          <circle
            cx="0"
            cy="0"
            r="20"
            stroke="#828282"
            stroke-opacity=".5"
            stroke-width="2"
          />
          <path
            d="M20 0c0-9.94-8.06-20-20-20"
          >
            <animatetransform
              attributeName="transform"
              dur="1s"
              from="0 0 0"
              repeatCount="indefinite"
              to="360 0 0"
              type="rotate"
            />
          </path>
        </g>
      </g>
    </svg>
  </div>
</DocumentFragment>
`;
